services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClient.java

```diff
@@ -69,6 +69,7 @@ private S3CrtAsyncHttpClient(Builder builder) {
                                  .withClientBootstrap(s3NativeClientConfiguration.clientBootstrap())
                                  .withPartSize(s3NativeClientConfiguration.partSizeBytes())
                                  .withComputeContentMd5(false)
+                                 .withMaxConnections(s3NativeClientConfiguration.maxConcurrency())
```

Comment:
It's not very straightforward to write a test to verify max connection behavior, so I tested it locally.